web/server/codechecker_server/api/report_server.py

```diff
@@ -1819,14 +1818,17 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
                 lambda x: x[0],
                 session.query(ReportAnnotations.key).distinct().all()))
 
-            annotation_cols = [
-                func.max(sqlalchemy.case([(
+            annotation_cols = OrderedDict()
+            for col in annotation_keys:
+                annotation_cols[col] = func.max(sqlalchemy.case([(
                     ReportAnnotations.key == col,
                     ReportAnnotations.value)])).label(f"annotation_{col}")
-                for col in annotation_keys]
 
-            is_unique = report_filter is not None and report_filter.isUnique
-            if is_unique:
+            if report_filter.isUnique:
```

Comment:
In the Thrift interface these functions get report filter as a required parameter, because it is not indicated as optional. So I added a default-constructed `ReportFilter()` object where it was `None` before.
By the way I am planning to configure `mypy` properly in the future so these type checks are done. I'm not convinced that `mypy` currently checks any of the type hints in the code.